e2e/cypress/tests/integration/performance/channel_switch_spec.js

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+// Stage: @prod
+// Group: @channel
+
+import {measurePerformance} from './utils.js';
+
+describe('Channel switch performance test', () => {
+    let testUser;
+
+    before(() => {
+        cy.apiInitSetup().then(({team, user}) => {
+            testUser = user;
+
+            // # Login as test user and go to town square
+            cy.apiLogin(testUser);
+            cy.visit(`/${team.name}/channels/town-square`);
+        });
+    });
+
+    it('measures switching between two channels from LHS', () => {
+        // # Invoke window object
+        measurePerformance('channelLoad', 5000, () => {
+            // # Switch channel to Off-topic
+
+            cy.get('#sidebarItem_off-topic').click({force: true});
```

Comment:
I think this was raised already, so not sure if solved in a different way.

On the first iteration, we have done the `before` step that brings us to the town square, and we move to off topic by doing this click.

But on every other iteration, we are already on off-topic, right? There will be no real channel switch.

Should we allow an optional parameter for `measurePerformance` that is something like "before each iteration"?

Reply:
Only one question related to a previous comment I made and one that @larkox mentioned above: should there be some kind of a reset of the state before each run, as after the first time the logic is ran, we'd already be in the off-topic channel? (e.g. a beforeEachRun that Daniel mentions or e.g. how the test is set up (switching from one channel to the other in the test case itself instead of clicking on off-topic, e.g. alternating between off-topic and town-square, so that the beforeEachRun function is not needed). Both approaches are similar for the current test suite though, so 0/5 on this.